lib/pine.coffee

```diff
@@ -51,19 +51,28 @@ getPine = ({ apiUrl, apiVersion, apiKey, request, auth } = {}) ->
 		# @todo Implement caching support.
 		###
 		_request: (options) ->
-			defaults options,
-				apiKey: apiKey
-				baseUrl: apiUrl
-				sendToken: !options.anonymous
+			auth.hasKey().then (hasKey) ->
+				authenticated = hasKey or !isEmpty(apiKey)
 
-			Promise.try ->
-				return if options.anonymous
+				defaults options,
+					apiKey: apiKey
+					baseUrl: apiUrl
+					sendToken: authenticated && !options.anonymous
 
-				auth.hasKey().then (hasKey) ->
-					if not hasKey and isEmpty(apiKey)
-						throw new errors.BalenaNotLoggedIn()
-			.then ->
-				return request.send(options).get('body')
+				request.send(options)
+					.get('body')
+					.catch statusCode: 401, (err) ->
+						# Always return the API error when the anonymous flag is used.
+						if options.anonymous
+							throw err
```

Comment:
It would be nice to avoid the sugar for performance
```suggestion
					.catch (err) ->
						if statusCode != 401
							throw err
						# Always return the API error when the anonymous flag is used.
						if options.anonymous
							throw err
```

Reply:
@Page-  that should be
```
if err.statusCode != 401
```
:P